### DISCUSSION TITLE
🔥🔥🔥 Hotfix Release 0.22.3 🔥🔥🔥

### DISCUSSION THREAD

**Discussion by peteeckel** [sys4/netbox-plugin-dns#181] [Announcements]
**Please upgrade to NetBox DNS version 0.22.3 immediately after or before you upgrade to NetBox 3.7.3**

NetBox 3.7.3 pushes the netaddr dependency from 0.10.1 to 1.2.1, which breaks a crucial functionality NetBox DNS was relying on. Data corruption within NetBox DNS and spurious errors can be the result.

If you notice strange behaviour with respect to reverse or RFC 2317 zones or the experimental IPAM Coupling feature, please contact me for a fix.

**Comment by bitcollector1** [comment]
So glad I saw this very early this morning after upgrading Netbox to 3.7.3!  Thanks again for being so organized that you were able to catch this issue so quickly! 

I upgraded thinking it would be safe since it was just an incremental update to NetBox. Next time I will not be so quick to upgrade to latest and greatest!

**Comment by peteeckel** [comment]
It would have happened even quicker had I not stumbled across > 70 failing tests in the NetBox 4.0 compatibility branch I'm working on. That caused me to review all changes in the NetBox `feature` branch before looking for the error in NetBox DNS' main branch, where it eventually could be found.

I had to trace back a while until I found that `arpa_to_prefix()` always returned `None`, leading to the `arpa_network` field for all zones that had been saved after the upgrade to 3.7.3 being `None` as well. That breaks the code that finds parent reverse zones, and that's not something you want to happen.

This can have really ugly consequences. Let's hope that it won't affect too many users. Everything can be fixed eventually, but it might be some effort depending on the case and how many changes were applied between the upgrade to NetBox 3.7.3 and NetBox DNS 0.22.3.
